src/ucp/core/ucp_mm.c

```diff
@@ -111,9 +116,35 @@ ucs_status_t ucp_mem_rereg_mds(ucp_context_h context, ucp_md_map_t reg_md_map,
             } else {
                 ucs_assert(address && length);
 
+                requested_reg_length  = length;
+                requested_reg_address = address;
+
+                if (md_attr->cap.reg_mem_types & UCS_BIT(mem_type)) {
+
+                    for (detect_md_index = 0; detect_md_index < context->num_mds;
+                         ++detect_md_index) {
+                        detect_md_attr = &context->tl_mds[detect_md_index].attr;
+                        if (detect_md_attr->cap.detect_mem_types & UCS_BIT(mem_type)) {
```

Comment:
we need to avoid overhead for memory registration on host memory. consider that most of the times we should go through the registration cache and memtype cache which are relatively fast, so would not want to pay the cost of another pointer detection

Reply:
Also thought about this concern and makes sense to not pay for this overhead every time registration is requested. As you suggested let's address this part in a separate PR.